krun/platform.py

```diff
@@ -0,0 +1,129 @@
+# Various CPU related abstrations
+
+import time
+import os
+import sys
+from collections import OrderedDict
+from krun.util import fatal, collect_cmd_output
+
+
+class BasePlatform(object):
+    CPU_TEMP_MANDATORY_WAIT = 1
+    CPU_TEMP_POLL_FREQ = 5
+    CPU_TEMP_POLLS_BEFORE_MELTDOWN = 12  #  * 5 = one minute
+
+    def __init__(self):
+        self.audit = OrderedDict()
+
+    def wait_until_cpu_cool(self):
+        time.sleep(BasePlatform.CPU_TEMP_MANDATORY_WAIT)
+        msg_shown = False
+        trys = 0
+        while True:
+            cool, reason = self.has_cpu_cooled()
+            if cool:
+                break
+
+            # if we get here, CPU is too hot!
+            if not msg_shown:
```

Comment:
I don't think they need a message immediately. In fact, it's likely to scare them. I agree if the pause was a minute or two they might get paranoid, but not if it's (say) 10-30s. For processes running multiple days, they'll be somewhat patient :)